Sources/hostmgr-helper/ManagedVirtualMachine.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+import Virtualization
+import OSLog
+import Network
+import libhostmgr
+
+@MainActor
+class ManagedVirtualMachine {
+    private let vmManager = VMManager()
+    private(set) var machine: VZVirtualMachine?
+    private(set) var ip: IPv4Address?
+
+    let config: LaunchConfiguration
+
+    var handle: String {
+        config.handle
+    }
+
+    init(config: LaunchConfiguration) {
+        self.config = config
+    }
+
+    func start() async throws {
+        Logger.helper.log("Start called for VM \(handle).")
+        do {
+            let newMachine = try await config.setupVirtualMachine()
+            self.machine = newMachine
+            try await newMachine.start()
+
+            if config.waitForNetworking {
+                self.ip = try await vmManager.ipAddress(forVmWithName: handle)
+                Logger.helper.log(
+                    "Startup of VM \(handle) complete – IP Address: \(ip.debugDescription)"
+                )
+            } else {
+                self.ip = .any
+                Logger.helper.log(
+                    "Startup of VM \(handle) in progress – skipped waiting for IP address per launch configuration"
+                )
+            }
+        } catch {
+            Logger.helper.error("Startup of \(handle) failed: \(error)")
+            await stop()
+            throw error
+        }
+    }
+
+    func stop() async {
```

Comment:
To be really robust this probably needs to protect against multiple calls - otherwise you could land in a situation where a VM is trying to stop while its bundle gets deleted.

I hesitate to introduce a new `Status` type but it may be necessary? Relying on the [VM state](https://developer.apple.com/documentation/virtualization/vzvirtualmachine/state-swift.property) is probably not enough.

Reply:
I did this in 4e6fb841588484a9b30a253871cc47d24a4858e7.